src/lib/ast.ml

```diff
@@ -1,31 +1,80 @@
+open Astring
+open Sexplib.Conv
+
 module DataFile = struct
-  type t = { id : int; path : string }
+  type t = { id : int; path : string; subpath : string option; wildcard : bool }
+  [@@deriving sexp]
+
+  let pp ppf t = Sexplib.Sexp.pp_hum ppf (sexp_of_t t)
+
+  let v ?(subpath = None) id path =
+    let wildcard =
+      match subpath with
+      | None -> false
+      | Some p ->
+          let last_char =
+            String.Sub.to_string (String.sub ~start:(String.length p - 1) p)
+          in
+          last_char = "*"
+    in
+    if wildcard then
+      {
+        id;
+        path = String.Sub.to_string (String.sub ~stop:(String.length path) path);
```

Comment:
Doesn't this produce the same string? Was the intent to drop the last character of the string?